<h2> Introduction to using Jupyter </h2>

This notebook is meant to give you an introduction to using Jupyter.

This notebook has a couple of questions that introduce you to the process of using Python code to answer questions, along with the particular syntax. If you're interested in learning more about the details of coding in Python, [this tutorial](https://www.w3schools.com/python/) is a great starting point.

Let's get started!

<h3> Part 1: Computing a sum with Python </h3>

In previous courses, you probability computed sums with sigma notation; for example,
$$\sum_{n = 1}^{1000} \frac{1}{n} = \frac 1 1 + \frac 1 2 + \frac 1 3 + \cdots \frac{1}{1000}.$$
These can get extremely cumbersome to compute by hand, but are easily handled by a computer using a [for loop](https://www.w3schools.com/python/python_for_loops.asp). The Python code that will compute this sum is below, with a lot of comments. Comments are placed on lines using \#, and are not interpreted as code.

In [1]:
# initialize the sum
s = 0


# run a for loop that adds 1/n each
for n in range(1, 1000 + 1):
    s = s + 1/n

print(s)


7.485470860550343


### Problem 1

So the sum of the first $1000$ numbers is about $7.49$. Now it's your turn; add a new code block to **answer the following two questions**.

* Compute $\sum_{n = 1}^{100} n^3$. Note that in Python, $a^b$ is written as `a**b`.

* Estimate $\sum_{n = 1}^{10^5} \frac{1}{n^2}.$

In [ ]:
# Put your answer to Problem 1a here

...

sum_of_cubes = ...
print(sum_of_cubes)

In [ ]:
# Put your answer to Problem 1b here

...

sum_of_squares = ...
print(sum_of_squares)

<h3> Part 2: Generating random numbers with Python </h3>

For this, we will use the [numpy](https://numpy.org) package, although there are many choices for generating random numbers using python.

Here is the documentation for the [randint()](https://numpy.org/doc/stable/reference/random/generated/numpy.random.randint.html) function we will be using.

In [14]:
import numpy as np
# choose either 0 or 1
flip = np.random.randint(low = 0 , high = 2)
print(flip)

0


For this problem, we will think of 0 as Tails and 1 as Heads

In [15]:
flips = np.random.randint(low = 0, high = 2, size = 100)
heads = flips.sum()
print(f'Number of heads = {heads}')

Number of heads = 58


### Problem 2

Suppose you flip $1000$ coins and record the number of heads.
Do this experiment $10$ times; what's the largest number of heads that you recorded in those ten runs?


In [ ]:
# Put your answer to Problem 2a here


largest_number_of_heads = ...
print(largest_number_of_heads)

How likely do you think it is to see a result of at least $600$ heads out of $1000$ flips? Do you think it's somewhat unlikely, very unlikely, or extremely unlikely?

In [16]:
# Put your answer to Problem 2b here

free_response = '...'

print(free_response)

...


<h3> Part 3: Submitting this to Gradescope </h3>

Once you've finished modifying your notebook and answering the questions, you'll need to submit it to Gradescope along with your other homework. To do this, generate a pdf file by clicking `File -> Download as... -> Notebook (.ipynb)`. Then upload that PDF to Gradescope and submit it to the assignment `Weekly Jupyter 1`. As always -- if you have any questions or run into any issues you can
* ask during discussion,
* email your TA or instructor,
* or bring them to office hours!